src/client/common/utils/async.ts

```diff
@@ -237,3 +239,22 @@ export async function flattenIterator<T>(iterator: AsyncIterator<T, void>): Prom
     }
     return results;
 }
+
+export class ChainedExecutions<T> {
```

Comment:
Could be deemed overkill to create a class, but i had to copy this code into two methods, hence decided to create a class...

Reply:
Class seems good to me.